eth/fetcher/block_fetcher.go

```diff
@@ -885,3 +890,23 @@ func (f *BlockFetcher) forgetBlock(hash common.Hash) {
 		delete(f.queued, hash)
 	}
 }
+
+func broadcastBlockEuclidV2(broadcastBlock blockBroadcasterFn, isEuclidV2 func(uint64) bool) blockBroadcasterFn {
+	return func(block *types.Block, propagate bool) {
+		block = block.CopyBlockDeepCopyHeader()
+		header := block.Header()
```

Comment:
`block.Header()` returns a copy so all modifications you're making here will be lost 

Same for other uses of `block.Header()`

Reply:
Deprecated. But in [the branch that explores this solution still](https://github.com/scroll-tech/go-ethereum/commit/1f96cf3083ab614419ad8bb1515253b3c72c03b5) this is solved with a new method.

This was pushed prematurely to inform of the direction of the decision (before it was finished).